src/sentry/api/endpoints/event_file_committers.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import (
+    Release, ReleaseCommit, Commit, CommitFileChange, Event
+)
+
+from itertools import izip
+from collections import defaultdict
+
+
+class EventFileCommittersEndpoint(ProjectEndpoint):
+
+    def _get_frame_paths(self, event):
+        data = event.data
+        try:
+            frames = data['sentry.interfaces.Stacktrace']['frames']
+        except KeyError:
+            try:
+                frames = data['sentry.interfaces.Exception']['values'][0]['stacktrace']['frames']
+            except KeyError:
+                return []  # can't find stacktrace information
+
+        return frames
+
+    def _score_path_match_length(self, pathA, pathB):
+        segmentsA = reversed(pathA.split('/'))  # todo maxbittker better splitting
+        segmentsB = reversed(pathB.split('/'))
+        score = 0
+        for a, b in izip(segmentsA, segmentsB):
+            if a != b:
+                break
+            score += 1
+        return score
+
+    def _get_commits(self, project_id, version):
+        try:
+            commits = Commit.objects.filter(
+                releasecommit=ReleaseCommit.objects.filter(
+                    release=Release.objects.get(
+                        projects=project_id,
+                        version=version,
+                    ),
+                )
+            )
+        except Commit.DoesNotExist:
+            return None
+        return commits
+
+    def _match_commits_frame(self, commits, frame):
```

Comment:
Actually, on second though, this is actually `O(n * m)` since this is called within another loop. We should figure out a way to rethink this logic a bit and try to do some primitive bulk fetching of data to normalize this.